enable student device to send question back to teachers device

symbyte opened the pull request:
Student devices can now return a question back to the teacher via inter device communication.
## Testing

One phone must be logged into a student account and the other must be logged into a teacher account.

Steps I took while testing are as follows:
1. On teacher phone, start roll call.
2. On student phone, go to roll call and check in.
3. On teacher phone, go to in class response and press the send button once, and clear the notification. (this step is needed in order to start @Perrofrijole's service, we will eventually be having our services start up on log in in a future update. We clear the notification so that, of course, we can see when the new question comes in.)
4. On student phone go to in class response, select an answer for the question, and press send.
5. Check the teachers device for a notification, it should be there.
6. The student device will notify the IP and port sent to when the ack is sent back to the student device. 

One issue that I ran into while testing is that, a few times, when trying to leave @song4u's fragment after checking in, I would get a crash from an exception. The exception would be saying something about trying to call the getResources object on a null reference. I assume this is happening because he might have some operation trying to be carried out after his fragment has been detached from the main activity. @song4u, I'm not sure if this is one of the known issue in you mentioned in your most recent pull request or not, so I thought it would be safest to mention it. Let me know if this is the case.

Commit:
enable student device to send question back to teachers device

wenxi-zeng commented:
No. That's something new. Thanks to mention it. I just had a test on it. I think it is because the fragment is trying to leave when the student roll call socket haven't get a sent result yet. I'll fix it tonight. For now, you can wait couple seconds after check-in. This should avoid the app to crash.

Commit:
remove connors call to start my service

this simply should not happen within his fragment.

symbyte commented:
When I rebased this onto the latest version of master something broke, working on a fix now.

Commit:
register connors service in the manafest

i guess when I was testing previously connor was starting my service along with his and his service wasnt actually getting started because it isnt registerd in the manifest. Fixed.

symbyte commented:
ok fixed it, it is ready for testing

colegarien commented:
Alright, will test now

colegarien committed:
Fix Crash due to Auto-Login

- Was comparing null with ""
- made so that if auth-token is not found it defaults to "" instead of null, this fixes the NullPointerException on API18

colegarien commented:
Ok, it appears to work as intended but there I don't see any notification as mentioned on 5 (other than the notification for opening the student activity)

It does seem there are a few crash that occur, but from what I can tell this is just from weird usage (like logging in and out as different users and such), we may think about a fail-safe operation of some sort that is called on logout that stops all services (P2P and android)

Also. the notification for the student is still appearing on the teacher's phone, shouldn't this be removed, and shouldn't the students phone get the notification?

symbyte commented:
The notification that appears on the teachers phone is similar to the one that I use on my service for the student to notify that a question arrives and launches my fragment. It is being used here as a placeholder just to show that the question indeed gets transmitted from the student device to the teacher device, and connor will be refactoring this in the future to instead bring up thomas' fragment to display the results of the question.

colegarien commented:
Ok, well I didn't see any major issues so I will merge unless there is anything else

symbyte commented:
Nothing on my end, merge away.
